session/session.go

```diff
@@ -851,6 +854,69 @@ func (s *session) SetGlobalSysVar(name, value string) error {
 	return errors.Trace(err)
 }
 
+//AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (s *session) AddGlobalBind(originSQL string, bindSQL string, defaultDB string, charset string, collation string) (err error) {
+	ctx := context.Background()
+
+	_, err = s.Execute(ctx, "BEGIN")
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() {
+		if err == nil {
+			_, err = s.execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := s.execute(ctx, "ROLLBACK")
+			terror.Log(errors.Trace(rbErr))
+		}
+	}()
+
+	originSQL = getEscapeCharacter(originSQL)
+	sql := fmt.Sprintf("SELECT status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		originSQL, defaultDB)
+	recordSet, err := s.execute(ctx, sql)
+	if err != nil {
+		return
+	}
+	if len(recordSet) == 1 {
+		var rows []chunk.Row
+		rows, err = drainRecordSet(ctx, s, recordSet[0])
+		if err != nil {
+			return
+		}
+		if len(rows) > 0 {
+			status := rows[0].GetString(0)
+			if status == bindinfo.BindUsing {
+				err = errors.New("origin sql already has binding sql")
+				return
+			}
+			sql = fmt.Sprintf("DELETE FROM mysql.bind_info WHERE original_sql='%s' and default_db='%s'", originSQL, defaultDB)
+			_, err = s.execute(ctx, sql)
+			if err != nil {
+				return
+			}
+		}
+	}
+
+	bindSQL = getEscapeCharacter(bindSQL)
+	ts := oracle.GetTimeFromTS(s.txn.StartTS())
+	sql = fmt.Sprintf(`INSERT INTO mysql.bind_info(original_sql,bind_sql,default_db,status,create_time,update_time,charset,collation) VALUES ('%s', '%s', '%s', '%s', '%s', '%s','%s', '%s')`,
+		originSQL, bindSQL, defaultDB, bindinfo.BindUsing, ts, ts, charset, collation)
+	_, err = s.execute(ctx, sql)
+	return
+}
+
+func getEscapeCharacter(str string) string {
```

Comment:
Why we need it?

Reply:
process sql like this:` select count(*) from test where DayID='43485';`

Reply:
Why not use the normalized sql?

Reply:
normalized sql will erase the bind sql's hint info. So we need a method to process escape character. Or we need a new normalize method which don't erase the hint info.